src/validations/css/stylelint.js

```diff
@@ -0,0 +1,57 @@
+import Validator from '../Validator';
+
+const errorMap = {
+  'syntaxError/Unclosed block': () => ({
+    reason: 'missing-closing-curly',
+  }),
+
+  'lintRule/declaration-block-trailing-semicolon': () => ({
+    reason: 'missing-semicolon',
+  }),
+};
+
+function isSyntaxError(error) {
+  return error.name === 'CssSyntaxError';
+}
+
+class StyleLintValidator extends Validator {
+  constructor(source) {
+    super(source, 'css', errorMap);
+  }
+
+  _getRawErrors() {
+    return System.import('../linters').then(
+      ({stylelint}) => stylelint(
+        this._source
+      ).then(
+        (result) => ({result}),
+        (syntaxError) => ({syntaxError})
+      )
+    );
+  }
+
+  _keyForError(error) {
+    if (isSyntaxError(error)) {
+      return `syntaxError/${error.reason}`;
+    }
+
+    return error.messages[0] && `lintRule/${error.messages[0].rule}`;
```

Comment:
One line that I think will look better once this method is processing a single discrete lint error : )